src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -245,6 +245,17 @@ class ConstraintsChecker
         return CheckConstraintMaxValue(itemName, current.Value(), static_cast<T>(expected));
     }
 
+    template <typename T, typename U, std::enable_if_t<!std::is_enum<T>::value, int> = 0>
+    bool CheckConstraintNotValue(const char * itemName, const chip::app::DataModel::Nullable<T> & current, const chip::app::DataModel::Nullable<U> & expected)
+    {
+        if (current == expected)
+        {
+            Exit(std::string(itemName) + " got unexpected value.");
```

Comment:
can we log what value we got and what was expected?

A log of "unexpecte value received" is much less debuggable than "Expected value of 123, got -1 instead" or "expected 100 got NULL instead".

Reply:
Updated.